homeassistant/components/homekit/type_sensors.py

```diff
@@ -139,13 +143,19 @@ def __init__(self, *args):
         """Initialize a AirQualitySensor accessory object."""
         super().__init__(*args, category=CATEGORY_SENSOR)
         state = self.hass.states.get(self.entity_id)
+        device_class = state.attributes.get(ATTR_DEVICE_CLASS)
```

Comment:
We need a separate class for each type as `density_to_air_quality` will only map PM25
```
def density_to_air_quality(density):
    """Map PM2.5 density to HomeKit AirQuality level."""
    if density <= 35:
        return 1
    if density <= 75:
        return 2
    if density <= 115:
        return 3
    if density <= 150:
        return 4
    return 5

```

Make a base class of `AirQualitySensor` and have the new types inhert it

Reply:
So we need to modify the homekit component together with homekit-controller component?

After your review, I think we need to do the following modification

1. create a pm10 and pm25 class that inhert the AirQualitySensor class on both ` homekit` and ` homekit-controller` component.
2. create a new density_to_air_quality mapping for pm10.


Am I right?